src/Executor/Values.php

```diff
@@ -110,16 +110,23 @@ public static function getVariableValues(Schema $schema, NodeList $varDefNodes,
                     } else {
                         // Otherwise, a non-null value was provided, coerce it to the expected
                         // type or report an error if coercion fails.
-                        $coerced = Value::coerceValue($value, $varType, $varDefNode);
+                        $coerced = Value::coerceInputValue($value, $varType);
 
                         $coercionErrors = $coerced['errors'];
                         if ($coercionErrors !== null) {
                             foreach ($coercionErrors as $error) {
-                                $invalidValue = Utils::printSafeJson($value);
+                                $invalidValue = $error->printInvalidValue();
+
+                                $inputPath = $error->printInputPath();
+                                $message = "Variable \"\${$varName}\" got invalid value {$invalidValue}"
+                                    . ($inputPath !== null
+                                        ? " at \"{$varName}{$inputPath}\""
+                                        : '')
+                                    . '; ' . $error->getMessage();
 
                                 $errors[] = new Error(
```

Comment:
I just came accross this - it actually replaces the error - shouldn't it be just 
```
$errors[] = $error;
```
rather than creating new error which hides the original one? or at least previous be `$error` not `$error->getPrevious()` ?